include/ginkgo/core/preconditioner/jacobi.hpp

```diff
@@ -338,7 +338,8 @@ class Jacobi : public EnableLinOp<Jacobi<ValueType, IndexType>>,
          *       has to be respected when setting this parameter. Failure to do
          *       so will lead to undefined behavior.
          */
-        gko::Array<index_type> GKO_FACTORY_PARAMETER(block_pointers, nullptr);
+        gko::Array<index_type> GKO_FACTORY_PARAMETER_SCALAR(block_pointers,
```

Comment:
Shouldn't this be considered a `_VECTOR`?

Reply:
You are probably right. We only use it as a `_SCALAR` throughout our code base, but I believe that the bug actually manifests itself based on the type of the parameter (primitive vs. composite with multiple-element constructor), so `_VECTOR` should be more appropriate.